components/sponsored_benefits/app/models/sponsored_benefits/services/general_agency_manager.rb

```diff
@@ -61,6 +61,21 @@ def set_default_general_agency
         prev_default_ga_id = current_default_ga.id if current_default_ga
         broker_agency_profile.default_general_agency_profile = general_agency_profile
         broker_agency_profile.save!
+        employers = SponsoredBenefits::Organizations::PlanDesignOrganization.where(owner_profile_id: form.broker_agency_profile_id, has_active_broker_relationship: true)
+        employers.each do |employer|
+          if employer.general_agency_accounts.first.present?
+            employer.general_agency_accounts.first.benefit_sponsrship_general_agency_profile_id = general_agency_profile.id
```

Comment:
this is assuming that the previous general agency is the first one, is it possible that is not the first one ? because the relationship is   embeds_many :general_agency_accounts, cascade_callbacks: true, validate: true

Reply:
I changed it to update_all, to update all the general agency accounts of an employer